analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -127,9 +138,15 @@ public void RegisterNodeAction<TSyntaxKind>(GeneratedCodeRecognizer generatedCod
         // the decision is made on based on whether the project contains the analyzer as NuGet).
         if (context.HasMatchingScope(supportedDiagnostics)
             && context.ShouldAnalyzeTree(sourceTree, generatedCodeRecognizer)
-            && LegacyIsRegisteredActionEnabled(supportedDiagnostics, sourceTree))
+            && LegacyIsRegisteredActionEnabled(supportedDiagnostics, sourceTree)
+            && ShoulAnalyzeRazorFile(sourceTree))
         {
             action(context);
         }
     }
+
+    private bool ShoulAnalyzeRazorFile(SyntaxTree sourceTree) =>
+        !GeneratedCodeRecognizer.IsRazorGeneratedFile(sourceTree)
+        || !supportedDiagnostics.Any(x => (x.CustomTags.Count() == 1 && x.CustomTags.Contains(DiagnosticDescriptorFactory.TestSourceScopeTag))
+                                          || rulesDisabledForRazor.Contains(x.Id));
```

Comment:
Educational: What is the CustomTag for?

Reply:
CustomTags are a place where you can store any additional information for any diagnostic descpritor (rule). We use it to store the information if the rule should target Main code, Test code or both.